list_api.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/EFForg/starttls-backend/db"
+	"github.com/EFForg/starttls-backend/policy"
+)
+
+////////////////////////////////
+//  *****   LIST API   *****  //
+////////////////////////////////
+
+func getNumberParam(r *http.Request, paramName string, defaultNumber int) int {
+	numStr, err := getParam(paramName, r)
+	result := defaultNumber
+	if err == nil {
+		if n, err := strconv.Atoi(numStr); err == nil {
+			result = n
+		}
+	}
+	return result
+}
+
+// GetList generates a new JSON file, with all queued entries!
+//   GET /auth/list
+//       expire_weeks: after how many weeks should this list expire? If unset
+//                     or invalid, defaults to 2. If set to 0, expires immediately.
+//       queued_weeks: for at least how many weeks should domains on the resulting list
+//                     have been queued? if unset or invalid, defaults to 1.
+func (api API) GetList(r *http.Request) APIResponse {
+	expireWeeks := getNumberParam(r, "expire_weeks", 2)
+	queuedWeeks := getNumberParam(r, "queued_weeks", 1)
+	list := api.List.Raw()
+	list.Timestamp = time.Now()
+	list.Expires = list.Timestamp.Add(time.Hour * 24 * 7 * time.Duration(expireWeeks))
```

Comment:
I'm not sure I understand what `list.Expires` is doing here. Is it just returning the current time plus the `expireWeeks` param to the user as JSON?

Reply:
Yep! To more properly motivate this endpoint, every week or so we release an update to the policy list, and it should have an updated timestamp/expiry date. These timestamps guarantee freshness to consumers of the list, as we can't provide many guarantees once the expiry date has passed.

So every week, in order to generate the "new" list, I've been using this branch to 1) update the timestamp to now and 2) set the expiry date to some time in the future. [Here's a sample](https://github.com/EFForg/starttls-everywhere/blob/master/RULES.md) of what the policy file looks like.

Reply:
Got it! I think when I reviewed this I didn't understand that it was meant to produce the actual list that will be distributed, and not just a convenient summary for you to do stuff with.